src/Controller/ContentTranslationPreviewController.php

```diff
@@ -132,23 +116,41 @@ public function previewRequestTitle(TranslationRequestInterface $oe_translation_
   /**
    * Builds the entity translation for the provided translation data.
    *
-   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
-   *   The entity for which the translation should be returned.
-   * @param array $data
-   *   The translation data for the fields.
+   * @param \Drupal\oe_translation\Entity\TranslationRequestInterface $oe_translation_request
+   *   The translation request.
    * @param string $language
-   *   The target language.
+   *   The language to preview in.
+   * @param string $bundle
+   *   The request bundle can be either 'local' or 'remote'.
    *
    * @return \Drupal\Core\Entity\ContentEntityInterface
    *   Translated entity.
    */
-  protected function getTranslationPreview(ContentEntityInterface $entity, array $data, string $language): ContentEntityInterface {
+  protected function getTranslationPreview(TranslationRequestInterface $oe_translation_request, string $language, string $bundle = 'local'): ContentEntityInterface {
+    // Get the entity being translated.
+    $entity = $oe_translation_request->getContentEntity();
+    if (!$entity) {
+      throw new \Exception('There is no entity currently being translated.');
+    }
+
+    // Retrieve the translation data.
+    if ($bundle !== 'local') {
+      $data = $oe_translation_request->getTranslatedData();
```

Comment:
What is this? `getTranslatedData ` is literally from another module. The entire point of this ticket is to move the logic of getting the data out of the entity where it belongs. By default it's `$oe_translation_request->getData();`, otherwise, the `$oe_translation_request->getTranslatedData();` belongs in the remote module

This block here defeats the entire purpose of the PR. You could have left everything exactly the same and just add this. No need for any other controller anything..